junit-platform-commons/src/main/java/org/junit/platform/commons/util/CollectionUtils.java

```diff
@@ -138,8 +140,12 @@ public static boolean isConvertibleToStream(Class<?> type) {
 				|| LongStream.class.isAssignableFrom(type)//
 				|| Iterable.class.isAssignableFrom(type)//
 				|| Iterator.class.isAssignableFrom(type)//
+				|| Spliterator.class.isAssignableFrom(type)//
 				|| Object[].class.isAssignableFrom(type)//
-				|| (type.isArray() && type.getComponentType().isPrimitive()));
+				|| (type.isArray() && type.getComponentType().isPrimitive())//
+				|| Arrays.stream(type.getMethods())//
+						.map(Method::getReturnType)//
+						.anyMatch(returnType -> returnType == Iterator.class || returnType == Spliterator.class));
```

Comment:
As hinted in https://github.com/junit-team/junit5/issues/3376#issuecomment-1625231605 we think this is too generic and should be more strict.

Reply:
I will adapt this to be more strict